### PR TITLE
PerformanceExplorer: Adds optional cell names to be constrained in pblock

### PR DESCRIPTION
**Pull request by clavin-xlnx**
Allows users to specify cell names to be constrained by pblock rather than the entire design.

### CODE REVIEW

File: src/com/xilinx/rapidwright/util/PerformanceExplorer.java [hunk 1]
```diff
@@ -314,12 +320,13 @@ public void explorePerformance(){
...
 		if(pblocks == null){
-			pblocks = new ArrayList<>();
-			pblocks.add(null);
+			pblocks = new HashMap<>();
+			pblocks.put(null, null);
 		}
 		
-		for(int pb=0; pb < pblocks.size(); pb++){
-			PBlock pblock = pblocks.get(pb);
+		int pb = 0;
```
Comment (eddieh-xlnx) [inline src/com/xilinx/rapidwright/util/PerformanceExplorer.java:327]: Doesn't look like this is used?

File: src/com/xilinx/rapidwright/util/PerformanceExplorer.java [hunk 2]
```diff
@@ -380,7 +388,7 @@ private static OptionParser createOptionParser(){
 			accepts(PLACER_DIRECTIVES_OPT).withOptionalArg().defaultsTo(placerDirectiveDefaults).describedAs("Comma separated list of place_design -directives");
 			accepts(ROUTER_DIRECTIVES_OPT).withOptionalArg().defaultsTo(routerDirectiveDefaults).describedAs("Comma separated list of route_design -directives");
 			accepts(CLK_UNCERTAINTY_OPT).withOptionalArg().describedAs("Comma separated list of clk uncertainty values (ns)");
-			accepts(PBLOCK_FILE_OPT).withRequiredArg().describedAs("PBlock file, one set of ranges per line");
+			accepts(PBLOCK_FILE_OPT).withRequiredArg().describedAs("PBlock file, one set of ranges per line with list of optional cell names to be constrained.  Use '|' after ranges to denote list of cell names separated by spaces");
```
Comment (eddieh-xlnx) [inline src/com/xilinx/rapidwright/util/PerformanceExplorer.java:391]: Appropriate to add an example? I can't quite imagine what the format is based on this description.

**Review by eddieh-xlnx — APPROVED** [review]

File: src/com/xilinx/rapidwright/util/PerformanceExplorer.java (hunk 1 above)
Comment (clavin-xlnx) [reply src/com/xilinx/rapidwright/util/PerformanceExplorer.java:327]: This is used as part of the `uniqueID`, (GitHub was hiding it from you).

File: src/com/xilinx/rapidwright/util/PerformanceExplorer.java (hunk 2 above)
Comment (clavin-xlnx) [inline src/com/xilinx/rapidwright/util/PerformanceExplorer.java:391]:
```suggestion
			accepts(PBLOCK_FILE_OPT).withRequiredArg().describedAs("PBlock file, one set of ranges per line with list of optional cell names to be constrained.  Use '|' after ranges to denote list of cell names separated by spaces, for example: 'SLICE_X0Y0:SLICE_X0Y1 DSP48E2_X0Y0:DSP48E2_X0Y1 | cell_name0 cell_name1'");
```

**Pull request merged**